netbox/core/models/jobs.py

```diff
@@ -224,6 +226,12 @@ def enqueue(cls, func, instance, name='', user=None, schedule_at=None, interval=
             job_id=uuid.uuid4()
         )
 
+        # Optionally, the job can be run immediately without being scheduled to run in the background.
+        if run_now:
+            func(job_id=str(job.job_id), job=job, **kwargs)
+            return job
```

Comment:
I could be missing something, but this seems like it defeats the purpose of using a job. Or if the goal is to reuse logic defined within a job, could we just call its `run()` method directly?

Reply:
My intention was to run a job in the foreground, but also create a `Job` object for documentation. Otherwise, jobs would run on the console, but there would be no audit trail. Now there's at least the information that a job was run, although the changelog objects are missing (maybe a topic for a future FR).

I agree that the method name `enqueue()` is a bit misleading. I could rename it to `run()` with the default parameter `background=True` to run jobs in the background, but also give the option to run it now. Since the `Job` API is not yet public for plugins, this should not break compatibility.